libraries/abstractions/common_io/test/test_iot_sdio.c

```diff
@@ -0,0 +1,1042 @@
+    testIotSdio_TPL_CODE_FUNCID, testIotSdio_TPL_CODE_FUNCE,
+};
+
+/*-----------------------------------------------------------*/
+
+/* Define Test Group. */
+TEST_GROUP( TEST_IOT_SDIO );
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Setup function called before each test in this group is executed.
+ */
+TEST_SETUP( TEST_IOT_SDIO )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Tear down function called after each test in this group is executed.
+ */
+TEST_TEAR_DOWN( TEST_IOT_SDIO )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief user callback function
+ */
+void prvSdioCallback( IotSdioSlotHandle_t const pxSdioHandle,
+                      IotSdioEventType_t eSdioEvent, void * pvUserContext )
+{
+    switch (eSdioEvent)
+    {
+        case eSdioCardInterruptEvent:
+            printf("sdiocallback with eSdioCardInterruptEvent\n");
+            break;
+        case eSdioCardInsertedEvent:
+            printf("sdiocallback with eSdioCardInsertedEvent\n");
+            break;
+        case eSdioCardRemovedEvent:
+            printf("sdiocallback with eSdioCardRemovedEvent\n");
+            break;
+        case eSdioCardPowerOnEvent:
+            printf("sdiocallback with eSdioCardPowerOnEvent\n");
+            break;
+        case eSdioCardPowerOffEvent:
+            printf("sdiocallback with eSdioCardPowerOffEvent\n");
+            break;
+        default:
+            printf("sdiocallback with unknown event\n");
+            break;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function to define which tests to execute as part of this group.
+ */
+TEST_GROUP_RUNNER( TEST_IOT_SDIO )
+{
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioOpenClose );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioDoubleOpenDoubleClose );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioOpenCloseFuzzing );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioCardConnectDisconnect );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioCardConnectDisconnectFuzzing );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioWriteReadDirect );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioWriteReadDirectFuzzing );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioWriteReadExtended );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioWriteReadExtendedFuzzing );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioIoctl );
+    RUN_TEST_CASE( TEST_IOT_SDIO, AFQP_IotSdioIoctlFuzzing );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test Function to test iot_sdio_open and iot_sdio_close.
+ *
+ */
+TEST( TEST_IOT_SDIO, AFQP_IotSdioOpenClose )
+{
+    IotSdioSlotHandle_t xSdioHandle = NULL;
+
+    int32_t lRetVal;
+
+    /* Open with valid host index and slot index should return a non NULL handle. */
+    xSdioHandle = iot_sdio_open(uctestIotSdioValidHostIdx,
```

Comment:
Can you use a global handle if present? If not open it? Please see I2C test as an example

Reply:
will update.

Reply:
done